### PR TITLE
Small cleanup around cookie rules

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs [hunk 1]
```diff
@@ -38,8 +38,8 @@ public sealed class CookieShouldBeHttpOnly : ObjectShouldBeInitializedCorrectlyB
                 KnownType.System_Web_HttpCookie,
                 KnownType.Microsoft_AspNetCore_Http_CookieOptions);
 
-        protected override CSharpObjectInitializationTracker ObjectInitializationTracker { get; } = new CSharpObjectInitializationTracker(
-            isAllowedConstantValue: constantValue => constantValue is bool value && value,
+        protected override CSharpObjectInitializationTracker ObjectInitializationTracker { get; } = new(
+            isAllowedConstantValue: constantValue => constantValue is true,
```
Comment (mary-georgiou-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs:42]: The `constantValue` is an object. Is it always a bool (thus there won't be an exception)?

**Review by mary-georgiou-sonarsource — CHANGES_REQUESTED** [review]
Just one comment regarding a type check.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs:42]: Is, in this case, will check both the type to be bool and the value to be true. It should behave just the same as the previous version, just that it's a bit more concise.

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**